### PR TITLE
[Duplicate of #5919]Data representation using brython

### LINKED ISSUE #5827: Data representation using brython
#### Proposed article introduction
When you hear `Brython`, it sounds like a made-up name or typography of Python. Brython is short for browser python. It is the implementation of python3 for Web development. Brython aims to replace Javascript as the language of the Web. Brython interacts with some of JavaScript's libraries to create projects they include; vue.js, phaser, bokeh, highcharts and, so on.

We aim to show readers how to represent data on the web using Brython.
#### Key takeaways 
It is a journey and, we want to take you along. We'll show the readers how to;
- Install the Brython module using the node package manager (npm).
- Create a piechart using Brython.
- Create a scatter plot using Brython.

#### Article quality
The article talks about the implementation of Python with the web without flask, rather a relatively unpopular library. 

#### References
N/A
### Conclusion
At the end of this article, the reader should be able to achieve the key takeaways listed above.

### PR DESCRIPTION
**Pull request by ayodele96**
> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5827 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by ayodele96** [comment]
@Bravinsimiyu this is the duplicate for the closed article

**Comment by ayodele96** [comment]
duplicate of #5919
cc: @srishilesh @Bravinsimiyu @hectorkambow

**Comment by Bravinsimiyu** [comment]
@srishilesh , could you assign me to this article

**Comment by hectorkambow** [comment]
Did the previous PR get a plagiairsm check?

**Comment by ayodele96** [comment]
@hectorkambow, yes it did

**Comment by Bravinsimiyu** [comment]
> Did the previous PR get a plagiairsm check?

Yes, it had passed the check

**Comment by srishilesh** [comment]
@Bravinsimiyu 

You may proceed with your review.

File: content/articles/data-representation-using-brython/index.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+To install the Highcharts module, use this code:
+
+```npm
+npm install highcharts
+```
+
+Doing this will download some important Highchart modules such as `highchart.js` and `exporting.js`. The exporting module allows you to download the charts in different file formats like PDF, PNG, SVG vector images, OR JPG.
+
+We will need to add `Jquery, exporting and highchart modules` to our HTML boilerplate. To do this, we'll refer to their respective file using the script tag below:
+
```
Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:101]:
```suggestion

Using the command above, we will download some important Highchart modules such as `highchart.js` and `exporting.js`. The `exporting.js` allows us to download the charts in different file formats like PDF, PNG, vector images(SVG), and JPG.

In our HTML boilerplate, we will add the  `Jquery` script, the `exporting.js` file, and the `highchart.js` module. Use the following code snippet:

```

File: content/articles/data-representation-using-brython/index.md [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+- `tooltip`: This object contains the details in the small box when we hover on our chart.
+- `allowPointSelect`: If set as `true`, it allows you to toggle or select any of the slices in the pie chart
+- `Series`: this is a set of data. In this case, it is the religious distribution of people in Nigeria.
+
+When we execute this code, our pie chart is as shown below:
+
+![piechart1](engineering-education/data-representation-with-react-and-ant-design/piechart1.png) 
+
+![piechart2](engineering-education/data-representation-with-react-and-ant-design/piechart2.png)
+
```
Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:216]: Please reorganize this section. This section is hard to follow. This means we have some steps that are missing or the steps are not well explained in chronological order. 

 For example, in beginning, you have said  "Having created our ` chart.py file`"... but there is no `chart.py` file we have created. You have created the `chart.py` file later on in the article.  First, create the `chart.py' file. Then you can import/add this file into your HTML boilerplate.  When creating the `chart.py` file make sure you explain the attributes/methods used.

Next, you can now start explaining the HTML file. Explain which files you have added to the HTML file like the `chart.py`,  `brython.js`, `brython_stdlib.js`, and other files. I believe this HTML file is the final part that holds all the imported files. 

You can not start explaining this HTML file and you have not explained how you have created the imported files.

Finally, you can now say that using the HTML file we will be able to create a pie chart.

Just ensure we have a chronologic order of the steps that a reader can easily follow.

File: content/articles/data-representation-using-brython/index.md [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+#### Creating a scatterplot using Brython.
+A scatterplot tests the relationship between two factors. To achieve this, we will put in place the HTML file used for the pie chart. The only difference is the code in the chart.py file.
+
```
Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:219]:
```suggestion
#### Creating a scatterplot using Brython.
A scatterplot tests the relationship between two factors. To achieve this, we will use the HTML file that we have previously used to create the pie chart. We only need to make some changes to the created chart.py file.

```

File: content/articles/data-representation-using-brython/index.md [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+##### Chart.py file
```
Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:220]:
```suggestion
##### Making changes to the Chart.py file
```

File: content/articles/data-representation-using-brython/index.md [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+The `chart.py` file for the scatterplot is as below:
```
Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:221]:
```suggestion
The new `chart.py` file that we will use to plot the scatterplot is shown below:
```

File: content/articles/data-representation-using-brython/index.md [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+![scatterplot](engineering-education/data-representation-with-react-and-ant-design/scatterplot.png)
+
+The code above plots the scatterplot using the `chart` object. The `chart` object has the following methods that specify the behavior of our scatterplot.
+
+- `type`: This method specifies the chart type. In this case, it is a scatterplot. Thus, it taking the value `scatter`
+- `zoomType`: This method specifies the zoom dimension when we move the mouse.
+- `radius`: This specifies the horizontal distance between the two closest points in a series.
+- `hover`: if enabled as true, hover takes effect when the mouse is placed over any of the series, thereby highlighting them.
+- `tooltip`: This is a small modal that pops up when we hover our mouse on the datasets in the plot. It displays all details added to the scatterplot.
+
```
Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:340]:
```suggestion
The code snippet above shows the relation between weight and height using a sample of 53 individuals of different races.

The code is used to plot the scatterplot using the `chart` object. The `chart` object has the following methods that specify the behavior of our scatterplot.
- `type`: This method specifies the chart type. In this case, it is a scatterplot. Thus, it taking the value `scatter`
- `zoomType`: This method specifies the zoom dimension when we move the mouse.
- `radius`: This specifies the horizontal distance between the two closest points in a series.
- `hover`: if enabled as true, hover takes effect when the mouse is placed over any of the series, thereby highlighting them.
- `tooltip`: This is a small modal that pops up when we hover our mouse on the datasets in the plot. It displays all details added to the scatterplot.

When we execute this code, the scatterplot is as shown below:

![scatterplot](engineering-education/data-representation-with-react-and-ant-design/scatterplot.png)
```

File: content/articles/data-representation-using-brython/index.md [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+                    },
+                },
+                "states": {"hover": {"marker": {"enabled": False}}},
+                "tooltip": {
+                    "headerFormat": "<b>{series.name}</b><br>",
+                    "pointFormat": "{point.x} cm, {point.y} kg",
+                },
+            }
+        },
+        "series": [
```
Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:250]: You can add the function of this `series` attribute when explaining this code snippet.

File: content/articles/data-representation-using-brython/index.md [hunk 8]
```diff
@@ -0,0 +1,345 @@
...
+
+- `type`: This method specifies the chart type. In this case, it is a scatterplot. Thus, it taking the value `scatter`
+- `zoomType`: This method specifies the zoom dimension when we move the mouse.
+- `radius`: This specifies the horizontal distance between the two closest points in a series.
+- `hover`: if enabled as true, hover takes effect when the mouse is placed over any of the series, thereby highlighting them.
+- `tooltip`: This is a small modal that pops up when we hover our mouse on the datasets in the plot. It displays all details added to the scatterplot.
+
+### Conclusion
+The process of data representation is a delightful journey. Like all journeys, it must come to an end.  Through the use of a Content Delivery Network, we were able to import Brython and Highchart's Javascript files. This further enabled us to represent data in the form of a pie chart and scatterplot using Brython. To get the Python code used in this tutorial, click [here](https://github.com/ayodele96/brython).
+
```
Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:343]:
```suggestion
Happy coding!
---
Peer Review Contributions by: [Bravin Wasike](/engineering-education/authors/bravin-wasike/)
```

**Review by Bravinsimiyu — CHANGES_REQUESTED** [review]
Hello @ayodele96, Overall nice work on adding more explanations on attributes and methods/attributes.

Commit the suggestions and also reorganize the highlighted section. Ensure it has chronological order of steps that a reader can easily follow.

Take your time when making the changes to ensure that everything is in place to avoid these back and forth changes. Ping me you are done so that we can move to the final review.

Thanks.

File: content/articles/data-representation-using-brython/index.md (hunk 3 above)
Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:219]: Please reorganize this section. This section is hard to follow. This means we have some steps that are missing or the steps are not well explained in chronological order.

For example, in beginning, you have said "Having created our chart.py file"... but there is no chart.py file we have created. You have created the chart.py file later on in the article. First, create the chart.py' file. Then you can import/add this file into your HTML boilerplate. When creating the chart.py` file make sure you explain the attributes/methods used.

Next, you can now start explaining the HTML file. Explain which files you have added to the HTML file like the chart.py, brython.js, brython_stdlib.js, and other files. I believe this HTML file is the final part that holds all the imported files.

You can not start explaining this HTML file and you have not explained how you have created the imported files.

Finally, you can now say that using the HTML file we will be able to create a pie chart.

Just ensure we have a chronologic order of the steps that a reader can easily follow.

Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:219]: Please reorganize this section. This section is hard to follow. This means we have some steps that are missing or the steps are not well explained in chronological order.

For example, in beginning, you have said "Having created our chart.py file"... but there is no chart.py file we have created. You have created the chart.py file later on in the article. First, create the chart.py' file. Then you can import/add this file into your HTML boilerplate. When creating the chart.py` file make sure you explain the attributes/methods used.

Next, you can now start explaining the HTML file. Explain which files you have added to the HTML file like the chart.py, brython.js, brython_stdlib.js, and other files. I believe this HTML file is the final part that holds all the imported files.

You can not start explaining this HTML file and you have not explained how you have created the imported files.

Finally, you can now say that using the HTML file we will be able to create a pie chart.

Just ensure we have a chronologic order of the steps that a reader can easily follow.

File: content/articles/data-representation-using-brython/index.md (hunk 7 above)
Comment (ayodele96) [reply content/articles/data-representation-using-brython/index.md:253]: I am trying to avoid repetitions. I already explained it in the piechart section

**Comment by ayodele96** [comment]
> Hello @ayodele96, Overall nice work on adding more explanations on attributes and methods/attributes.
> 
> Commit the suggestions and also reorganize the highlighted section(section from `####Adding the brython.js and brython_stdlib.js file using our script tag.` to where you create the pie chart. Ensure it has chronological order of steps that a reader can easily follow.
> 
> Take your time when making the changes to ensure that everything is in place to avoid these back and forth changes. Ping me you are done so that we can move to the final review.
> 
> Thanks.

I added that section separately since both the piechart and the scatterplot will use the same boilerplate. I don't think there's a need to move it.

File: content/articles/data-representation-using-brython/index.md (hunk 3 above)
Comment (ayodele96) [reply content/articles/data-representation-using-brython/index.md]: effected this changes

File: content/articles/data-representation-using-brython/index.md [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+- `tooltip`: This object contains the details in the small box when we hover on our chart.
+- `allowPointSelect`: If set as `true`, it allows you to toggle or select any of the slices in the pie chart
+- `Series`: this is a set of data. In this case, it is the religious distribution of people in Nigeria.
+
+When we execute this code, our pie chart is as shown below:
+
+![piechart1](engineering-education/data-representation-with-react-and-ant-design/piechart1.png) 
+
+![piechart2](engineering-education/data-representation-with-react-and-ant-design/piechart2.png)
+
```
Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:224]:
```suggestion

![piechart1](engineering-education/data-representation-using-brython/piechart1.png) 

![piechart2](engineering-education/data-representation-using-brython/piechart2.png)

```

Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:224]:
```suggestion

![piechart1](engineering-education/data-representation-using-brython/piechart1.png) 

![piechart2](engineering-education/data-representation-using-brython/piechart2.png)

```

Comment (Bravinsimiyu) [reply content/articles/data-representation-using-brython/index.md:224]: Why is there a need for using 2 piechart diagrams that show the same information? Please remove the redundant one. And rename the remaining one from `piechart1` to `piechart`. Also, do that the image name.

File: content/articles/data-representation-using-brython/index.md [hunk 10]
```diff
@@ -0,0 +1,353 @@
...
+The code is used to plot the scatterplot using the `chart` object. The `chart` object has the following methods that specify the behavior of our scatterplot.
+- `type`: This method specifies the chart type. In this case, it is a scatterplot. Thus, it taking the value `scatter`
+- `zoomType`: This method specifies the zoom dimension when we move the mouse.
+- `radius`: This specifies the horizontal distance between the two closest points in a series.
+- `hover`: if enabled as true, hover takes effect when the mouse is placed over any of the series, thereby highlighting them.
+- `tooltip`: This is a small modal that pops up when we hover our mouse on the datasets in the plot. It displays all details added to the scatterplot.
+
+When we execute this code, the scatterplot is as shown below:
+
+![scatterplot](engineering-education/data-representation-with-react-and-ant-design/scatterplot.png)
```
Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:346]:
```suggestion
![Scatterplot](engineering-education/data-representation-using-brython/scatterplot.png)
```

File: content/articles/data-representation-using-brython/index.md [hunk 11]
```diff
@@ -0,0 +1,353 @@
...
+### Conclusion
+The process of data representation is a delightful journey. Like all journeys, it must come to an end.  Through the use of a Content Delivery Network, we were able to import Brython and Highchart's Javascript files. This further enabled us to represent data in the form of a pie chart and scatterplot using Brython. To get the Python code used in this tutorial, click [here](https://github.com/ayodele96/brython).
+Happy coding!
+---
+Peer Review Contributions by: [Bravin Wasike](/engineering-education/authors/bravin-wasike/)
+### Further Reading
+For further reading on Brython, read this [documentation](https://brython.info/static_doc/en/intro.html).
```
Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:353]:
```suggestion
### Conclusion
The process of data representation is a delightful journey. Like all journeys, it must come to an end. Through the use of a Content Delivery Network, we were able to import Brython and Highchart's Javascript files. This further enabled us to represent data in the form of a pie chart and scatterplot using Brython. 

To get the Python code used in this tutorial, click [here](https://github.com/ayodele96/brython).

### Further Reading
For further reading on Brython, read this [documentation](https://brython.info/static_doc/en/intro.html).

I hope you find this article helpful.

Happy coding!

---
Peer Review Contributions by: [Bravin Wasike](/engineering-education/authors/bravin-wasike/)
```

**Review by Bravinsimiyu — CHANGES_REQUESTED** [review]
Hello @ayodele96, Thanks for making the changes. Your article has now a good flow and it's easy to follow. For future reference avoid using `we'll` instead use `we will` also avoid using `let us` instead use `let's`.

Commit the few changes and your article will be ready for final which will be done by the content moderators. Also, ensure you preview the article to see how it renders on the browser. While in VS Code you can use `ctr+shift+V` to preview the article.

**Comment by ayodele96** [comment]
@Bravinsimiyu, I have effected all the changes and I think we are good to go🚀

File: content/articles/data-representation-using-brython/index.md [hunk 12]
```diff
@@ -0,0 +1,359 @@
...
+- `plotBorderWidth`: It plots the width of the chart area in pixels.
+- `plotShadow`: This method takes in boolean values and adds a drop shadow to the area. 
+- `renderTo`: This method is like `document.getElementBy` in javascript. It will take the `id` of the `div` tag in our HTML code.
+- `tooltip`: This object contains the details in the small box when we hover on our chart.
+- `allowPointSelect`: If set as `true`, it allows you to toggle or select any of the slices in the pie chart
+- `Series`: this is a set of data. In this case, it is the religious distribution of people in Nigeria.
+
+When we execute this code, our pie chart is as shown below:
+
+![piechart1](engineering-education/data-representation-using-brython/piechart1.png)
```
Comment (Bravinsimiyu) [inline content/articles/data-representation-using-brython/index.md:221]:
```suggestion
![Piechart](engineering-education/data-representation-using-brython/piechart1.png) 
```

Comment (Bravinsimiyu) [reply content/articles/data-representation-using-brython/index.md:221]: Make that change and your article will be ready for final review.

**Comment by ayodele96** [comment]
done

**Comment by Bravinsimiyu** [comment]
Hello @srishilesh, This article is ready for a final review
cc @hectorkambow @ahmadmardeni1

**Comment by ayodele96** [comment]
@srishilesh and @hectorkambow, I have added the front matter  and the tech review has been done

**Comment by WanjaMIKE** [comment]
Merged #6156  into master
